src/librustc_resolve/lib.rs

```diff
@@ -1195,6 +1195,12 @@ pub struct Resolver<'a> {
     pub whitelisted_legacy_custom_derives: Vec<Name>,
     pub found_unresolved_macro: bool,
 
+    // List of macros that we need to warn about as being unused.
+    // The bool is true if the macro is unused, and false if its used.
+    // Setting a bool to false should be much faster than removing a single
+    // element from a FxHashSet.
+    unused_macros: FxHashMap<DefId, bool>,
```

Comment:
I still think this should be an `FxHashSet<DefId>` -- I believe the perf difference is negligible here (removing a value from a hash set is already many orders of magnitude faster than the rest the processing we do per used crate-local `macro_rules!`).